ocaml/gencert/selfcert.ml

```diff
@@ -74,60 +78,90 @@ let call_openssl args =
   in
   Forkhelpers.execute_command_get_output openssl ~env args
 
-(** [generate_private_key] calls openssl to generate an RSA key of
-  [length] bits. *)
-let generate_private_key length =
+(** [generate_pub_priv_key] calls openssl to generate an RSA key of
+    [length] bits. *)
+let generate_pub_priv_key length =
   let args = ["genrsa"; string_of_int length] in
-  let stdout, _stderr = call_openssl args in
-  stdout
-
-let selfsign issuer extensions key_length days certfile =
-  let rsa =
+  let* rsa_string =
     try
-      generate_private_key key_length
-      |> Cstruct.of_string
-      |> X509.Private_key.decode_pem
-      |> R.failwith_error_msg
-      |> function
-      | `RSA x ->
-          x
+      let stdout, _stderr = call_openssl args in
+      Ok stdout
     with e ->
-      let msg =
-        Printf.sprintf "generating RSA key for %s failed: %s" certfile
-          (Printexc.to_string e)
-      in
-      D.error "%s" msg ; failwith msg
+      let msg = "generating RSA key failed" in
+      D.error "selfcert.ml: %s" msg ;
```

Comment:
You could use `__LOC__` or `__FUNCTION__`

Reply:
I didn't know about `__FUNCTION__`, but unfortunately that is 4.12 :(